CI Failure on Py3.9 PySide6

## 🐛 Bug

I'm seeing this failure on #5216, and can't find any issues reporting it. The test suite fails at test discovery time with the following traceback:

<details>

<summary>

```pytb
  E   ValueError: <enum 'NotificationSeverity'> may only be called with a `str` or an instance of <enum 'NotificationSeverity'>. Got <class 'shibokensupport.enum_310.auto'>
```

</summary>

```pytb
  ________________________ ERROR collecting test session _________________________
  /opt/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/importlib/__init__.py:127: in import_module
      return _bootstrap._gcd_import(name[level:], package, level)
  <frozen importlib._bootstrap>:1030: in _gcd_import
      ???
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/conftest.py:51: in <module>
      from napari.components import LayerList
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/components/__init__.py:19: in <module>
      from .layerlist import LayerList
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/components/layerlist.py:9: in <module>
      from ..layers import Layer
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/layers/__init__.py:12: in <module>
      from .base import Layer
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/layers/base/__init__.py:1: in <module>
      from .base import Layer, no_op
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/layers/base/base.py:28: in <module>
      from ...utils.key_bindings import KeymapProvider
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/utils/key_bindings.py:46: in <module>
      from ..settings import get_settings
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/settings/__init__.py:6: in <module>
      from ._napari_settings import CURRENT_SCHEMA_VERSION, NapariSettings
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/settings/_napari_settings.py:10: in <module>
      from ._application import ApplicationSettings
  <frozen importlib._bootstrap>:1007: in _find_and_load
      ???
  <frozen importlib._bootstrap>:986: in _find_and_load_unlocked
      ???
  <frozen importlib._bootstrap>:680: in _load_unlocked
      ???
  /tmp/.tox/py39-linux-pyside6/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
      exec(co, module.__dict__)
  napari/settings/_application.py:20: in <module>
      class ApplicationSettings(EventedModel):
  napari/utils/events/evented_model.py:82: in __new__
      cls = super().__new__(mcs, name, bases, namespace, **kwargs)
  pydantic/main.py:198: in pydantic.main.ModelMetaclass.__new__
      ???
  pydantic/fields.py:506: in pydantic.fields.ModelField.infer
      ???
  pydantic/fields.py:436: in pydantic.fields.ModelField.__init__
      ???
  pydantic/fields.py:546: in pydantic.fields.ModelField.prepare
      ???
  pydantic/fields.py:570: in pydantic.fields.ModelField._set_default_and_type
      ???
  pydantic/fields.py:439: in pydantic.fields.ModelField.get_default
      ???
  pydantic/utils.py:695: in pydantic.utils.smart_deepcopy
      ???
  /opt/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/copy.py:172: in deepcopy
      y = _reconstruct(x, memo, *rv)
  /opt/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/copy.py:264: in _reconstruct
      y = func(*args)
  napari/utils/misc.py:279: in __call__
      raise ValueError(
  E   ValueError: <enum 'NotificationSeverity'> may only be called with a `str` or an instance of <enum 'NotificationSeverity'>. Got <class 'shibokensupport.enum_310.auto'>
```

</details>

## To Reproduce

Here is a link to the failed run:

https://github.com/napari/napari/actions/runs/3247412851/jobs/5327346396#step:8:187

I'm not sure yet whether it is sporadic...

--- a/.github/workflows/test_pull_requests.yml
+++ b/.github/workflows/test_pull_requests.yml
@@ -80,10 +80,6 @@ jobs:
           - python: 3.9
             platform: ubuntu-latest
             backend: pyside6
-# uncoment when new qtpy is released
-#          - python: '3.10'
-#            platform: ubuntu-latest
-#            backend: pyside6
 
     steps:
       - name: Cancel Previous Runs

--- a/docs/release/release_0_4_17.md
+++ b/docs/release/release_0_4_17.md
@@ -44,6 +44,8 @@ See the docstring of `napari.imshow` for more details.
 
 We added some basic support for PyQt6 and PySide6 in #3707, though we expect some issues.
 If you need or want to use Qt6, please try this out and report any bugs you find.
+If you plan to use PySide6 specifically, please use Python 3.10 to avoid issues with older
+versions of Python.
 
 ### Multi-color text
 

--- a/napari/__main__.py
+++ b/napari/__main__.py
@@ -436,10 +436,6 @@ def _maybe_rerun_with_macos_fixes():
        This requires relaunching the app from a symlink to the
        desired python executable, conveniently named 'napari'.
     """
-    from napari._qt import API_NAME
-
-    # This import mus be here to raise exception about PySide6 problem
-
     if sys.platform != "darwin":
         return
 
@@ -454,6 +450,8 @@ def _maybe_rerun_with_macos_fixes():
     import platform
     import subprocess
     from tempfile import mkdtemp
+
+    from qtpy import API_NAME
 
     # In principle, we will relaunch to the same python we were using
     executable = sys.executable

--- a/napari/_qt/__init__.py
+++ b/napari/_qt/__init__.py
@@ -6,7 +6,7 @@ from warnings import warn
 from ..utils.translations import trans
 
 try:
-    from qtpy import API_NAME, QT_VERSION, QtCore
+    from qtpy import API_NAME, QtCore
 except Exception as e:
     if 'No Qt bindings could be found' in str(e):
         raise type(e)(
@@ -26,17 +26,6 @@ if API_NAME == 'PySide2':
     os.environ['QT_PLUGIN_PATH'] = str(
         Path(PySide2.__file__).parent / 'Qt' / 'plugins'
     )
-
-if API_NAME == 'PySide6' and sys.version_info[:2] < (3, 10):
-    from packaging import version
-
-    if version.parse(QT_VERSION) > version.parse("6.3.1"):
-        raise RuntimeError(
-            trans._(
-                "Napari is not expected to work with PySide6 >= 6.3.2 on Python < 3.10",
-                deferred=True,
-            )
-        )
 
 
 # When QT is not the specific version, we raise a warning:

--- a/napari/_vispy/__init__.py
+++ b/napari/_vispy/__init__.py
@@ -1,11 +1,10 @@
 import logging
 
+import qtpy
 from vispy import app
 
-from napari._qt import API_NAME
-
 # set vispy application to the appropriate qt backend
-app.use_app(API_NAME)
+app.use_app(qtpy.API_NAME)
 del app
 
 # set vispy logger to show warning and errors only

--- a/tox.ini
+++ b/tox.ini
@@ -73,9 +73,9 @@ setenv =
 deps =
     pytest-cov
     pyqt6: PyQt6
-    pyside6: PySide6 < 6.3.2 ; python_version < '3.10'
-    pyside6: PySide6 ; python_version >= '3.10'
+    pyside6: PySide6
     pytest-json-report
+    pyside6: shiboken6!=6.3.2
 # use extras specified in setup.cfg for certain test envs
 extras =
     testing

--- a/tox.ini
+++ b/tox.ini
@@ -70,12 +70,15 @@ setenv =
     async: NAPARI_ASYNC = 1
     async: PYTEST_ADDOPTS = --async_only
     async: PYTEST_PATH = napari
+# PySide6 6.4.0 broke Python 3.9 and under.
+# See the napari issue for more details:
+# https://github.com/napari/napari/issues/5217
 deps =
     pytest-cov
     pyqt6: PyQt6
-    pyside6: PySide6
+    pyside6: PySide6 < 6.3.2 ; python_version < '3.10'
+    pyside6: PySide6 ; python_version >= '3.10'
     pytest-json-report
-    pyside6: shiboken6!=6.3.2
 # use extras specified in setup.cfg for certain test envs
 extras =
     testing

